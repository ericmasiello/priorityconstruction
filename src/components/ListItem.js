import React from 'react';
import styled from 'styled-components';
import * as CustomPropTypes from '../propTypes';

export const ListItem = ({ tag: Tag, ...rest }) => <Tag {...rest} />;

ListItem.propTypes = {
  tag: CustomPropTypes.Tag,
};

ListItem.defaultProps = {
  tag: 'li',
};

ListItem.displayName = 'ListItem';

const StyleListItem = styled(ListItem)`
  &:not(:last-child) {
    margin-bottom: 0.5rem;
  }
`;

export default StyleListItem;
