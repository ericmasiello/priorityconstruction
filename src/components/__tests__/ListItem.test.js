import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import ListItem, { ListItem as RawListItem } from '../ListItem';

it('should render <ListItem>', () => {
  const component = renderer.create(<ListItem />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render children', () => {
  const wrapper = shallow(<RawListItem>Test</RawListItem>);

  expect(wrapper.text()).toBe('Test');
});

it('should render additional props', () => {
  const wrapper = shallow(<RawListItem data-test="my-attribute" />);

  expect(wrapper.find('[data-test="my-attribute"]')).toHaveLength(1);
});

it('should display additional classes', () => {
  const wrapper = shallow(<RawListItem className="my-class" />);

  expect(wrapper.hasClass('my-class')).toBe(true);
});

it('should render as a <li> by default', () => {
  const wrapper = shallow(<RawListItem />);

  expect(wrapper.type()).toBe('li');
});

it('should render with a custom tag', () => {
  const wrapper = shallow(<RawListItem tag="span" />);

  expect(wrapper.type()).toBe('span');
});
