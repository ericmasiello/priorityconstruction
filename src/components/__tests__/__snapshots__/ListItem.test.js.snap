// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <ListItem> 1`] = `
.c0:not(:last-child) {
  margin-bottom: 0.5rem;
}

<li
  className="c0"
/>
`;
